tests/javascript/frontediting/spec.js

```diff
@@ -0,0 +1,208 @@
+ * @version     1.0.0
+ */
+
+define(['jquery', 'testsRoot/frontediting/spec-setup', 'jasmineJquery'], function ($) {
+
+	describe('Frontediting jEditMakeAbsolute with argument true', function () {
+		beforeAll(function () {
+			$('.fr-ed-make-abs-true').jEditMakeAbsolute(true);
+		});
+
+		it('Should change style of all selected elements ', function () {
+			var $elements = $('.fr-ed-make-abs-true');
+			var $first = $elements.first();
+			var $second = $elements.last();
+
+			expect($first).toHaveCss({position: "absolute", marginLeft: '0px', marginTop: '0px'});
+			expect($second).toHaveCss({position: "absolute", marginLeft: '0px', marginTop: '0px'});
+		});
+
+		it('Should detach all selected elements from jEditMakeAbsolute div', function () {
+			expect($('#jEditMakeAbsolute')).not.toContainElement('.fr-ed-make-abs-true');
+		});
+
+		it('Should append all detached elements to body', function () {
+			expect($('body>div.fr-ed-make-abs-true').length).toEqual(2);
+		});
+	});
+
+	describe('Frontediting jEditMakeAbsolute with argument false', function () {
+		beforeAll(function () {
+			$('.fr-ed-make-abs-false').jEditMakeAbsolute(false);
+		});
+
+		it('Should change style of all selected elements ', function () {
+			var $elements = $('.fr-ed-make-abs-false');
+			var $first = $elements.first();
+			var $second = $elements.last();
+
+			expect($first).toHaveCss({position: "absolute", marginLeft: '0px', marginTop: '0px'});
+			expect($second).toHaveCss({position: "absolute", marginLeft: '0px', marginTop: '0px'});
+		});
+
+		it('Should not detach any selected elements from jEditMakeAbsolute div', function () {
+			expect($('#jEditMakeAbsolute')).toContainElement('.fr-ed-make-abs-false');
+		});
+	});
+
+	describe('Frontediting modules jmoddiv on mouseenter', function () {
+		beforeAll(function () {
+			$('#jmoddiv-module').mouseenter();
+			this.$editBtn = $('body>a.btn.jmodedit');
+		});
+
+		it('Should add class jmodinside to the div', function () {
+			expect($('#jmoddiv-module')).toHaveClass('jmodinside');
+		});
+
+		it('Should prepend edit icon element and move it to end of body', function () {
+			expect(this.$editBtn).toExist();
+		});
+
+		it('Should set jmodedit class on the added element', function () {
+			expect(this.$editBtn).toHaveClass('jmodedit');
+		});
+
+		it('Should set attribute href = "url" on the added element', function () {
+			expect(this.$editBtn).toHaveAttr('href', 'url');
+		});
+
+		it('Should set attribute title = "title" on the added element', function () {
+			expect(this.$editBtn).toHaveAttr('data-original-title', 'mod-title');
+		});
+
+		it('Should set attribute target = "target" on the added element', function () {
+			expect(this.$editBtn).toHaveAttr('target', 'target');
+		});
+
+		describe('Edit icon on mouseenter', function () {
+			beforeAll(function () {
+				this.$editBtn.mouseenter();
+			});
+
+			it('Should show tooltip', function () {
+				expect($('body>div.tooltip')).toExist();
```

Comment:
yeah but here I wanted to specifically test if the specified element is a direct child of body. Is there any better way that can be checked?